### ISSUE TITLE
enum input arguments are capitalized

### ISSUE THREAD

**Issue by mikestaub** [IBM/openapi-to-graphql#266]
**Describe the bug**

After posting a mutation that accepts an enum field as input, the underlying OpenAPI service receives a variable with the first letter capitalized. 

Is this intentional? If so, how can I disable this feature?

**Expected behavior**

I expect all the input variables to maintain their casing.

**Comment by Alan-Cha** [comment]
This sounds like a bug. To clarify, you mean the mutation uses an input object type with a field that is an enum and the variable name (not the variable value) in the API call is unexpected capitalized?

**Comment by mikestaub** [comment]
Exactly. After further debugging it seems this is because the OpenAPI type is Capitalized. If I change it to lower case, the variable is lowercased.

**Comment by mikestaub** [comment]
Here is a branch with failing tests.

https://github.com/mikestaub/openapi-to-graphql/pull/1

I suspect the fix has to do with `addObjectPropertiesToDataDef()`

**Comment by Alan-Cha** [comment]
@mikestaub Thank you so much for creating this branch. This branch was invaluable for debugging this problem.

As you may be aware of, OtG sanitizes strings from the OAS and then reverses the process where necessary in order to make the appropriate call. I believe this issue is the result of improper reversal.

The method we currently use is, honestly, a bit primitive. The summarize, we have a map named [`saneMap`](https://github.com/IBM/openapi-to-graphql/blob/a8203694adbf26e78f1f64bb3d1991a34c1b8094/packages/openapi-to-graphql/src/preprocessor.ts#L38) that correlates sanitized strings with their raw versions. There are a few places where we store strings but most of time, it happens [here](https://github.com/IBM/openapi-to-graphql/blob/a8203694adbf26e78f1f64bb3d1991a34c1b8094/packages/openapi-to-graphql/src/preprocessor.ts#L477), where we recursively analyze the different schemas that are used in the OAS. 

This current method is a problem because we arbitrarily dumping all of the sanitized and raw strings into the same place without making any effort to consider their contexts. Therefore, when we actually try to desanitize the strings [here](https://github.com/IBM/openapi-to-graphql/blob/a8203694adbf26e78f1f64bb3d1991a34c1b8094/packages/openapi-to-graphql/src/resolver_builder.ts#L246), we encounter problems. 

We went with this method because the "right" way to do it is complex and this was a more reasonable solution during our prototyping phase. I think the right way to do it would be to create a map for GraphQL (input) object type names, and multiple maps for field names in each of those GraphQL (input) object types. 

However, I'm not sure if I have the time to make all the changes needed. If you would like to help, or anyone else, I would be happy to assist. I think for now, we can at least create two separate maps for GraphQL (input) object type names and one for field names. I believe this strategy will solve a significant percentage of possible conflicts by itself. 

The error we are getting from your branch would be fixed by this solution. In the error, the field `address` is mistakenly "desanitized" to `Address` which corresponds to a GraphQL object type name. In other words, field names are getting mixed up with object type names. If you could check to see if the enum fields in your GraphQL interface also corresponds to similarly named GraphQL object types, then I think that would be more confirmation for this solution.

**Comment by mikestaub** [comment]
"I think the right way to do it would be to create a map for GraphQL (input) object type names, and multiple maps for field names in each of those GraphQL (input) object types." I agree with this.

I would be happy to take this on, but unfortunately, I don't have any free time in the coming weeks, and we have an acceptable workaround. We simply loop through all the definitions and lowercase the first letter.
